### PR TITLE
HYDRA-909 - remove lighted prefix from hydra scene hierarchy

### PR DESCRIPTION
**Pull request by lilike-adsk**
To remove "lighted" prefix from hydra scene hierarchy, we collect all prims that need lighting/shadowing and put them in a SdfPathVector, and set to shadowCollection.

Impact: when shadow is on, modification to the scene might be slow a bit.
Peformance testing (adding prim into a scene with 10000 prims when shadow is on and tumbling scene with 10000 prims when shadow is on) shows the impact is trivial.

### CODE REVIEW

File: lib/mayaHydra/hydraExtensions/adapters/lightAdapter.cpp [hunk 1]
```diff
@@ -154,12 +154,7 @@ VtValue MayaHydraLightAdapter::Get(const TfToken& key)
         MFnLight       mayaLight(GetDagPath());
         GlfSimpleLight light;
         const auto     color = mayaLight.color();
-        auto     intensity   = mayaLight.intensity();
```
Comment (lanierd-adsk) [inline lib/mayaHydra/hydraExtensions/adapters/lightAdapter.cpp]: You probably don't have the up to date file as this shouldn't be removed, unless you moved it somewhere else ?

Comment (lilike-adsk) [reply lib/mayaHydra/hydraExtensions/adapters/lightAdapter.cpp]: yes, need merge with latest

Comment (lilike-adsk) [reply lib/mayaHydra/hydraExtensions/adapters/lightAdapter.cpp]: @lanierd-adsk you can wait until it's ready, still need to update tests and polish, and do performance tests. :) will let you know when it's ready

Comment (lanierd-adsk) [reply lib/mayaHydra/hydraExtensions/adapters/lightAdapter.cpp]: Sounds good. Thanks.

File: lib/flowViewport/sceneIndex/fvpWireframeSelectionHighlightSceneIndex.cpp [hunk 2]
```diff
@@ -333,6 +333,11 @@ WireframeSelectionHighlightSceneIndex(
     TF_AXIOM(_wireframeColorInterface);
 
     auto operation = [this](const SdfPath& primPath, const HdSceneIndexPrim& prim) -> bool {
+
+        if (_IsExcluded(primPath)) {
```
Comment (lilike-adsk) [inline lib/flowViewport/sceneIndex/fvpWireframeSelectionHighlightSceneIndex.cpp:337]: Grab the fix from Pierre, as removing Lighted hierarcky will expose this potential issue

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp [hunk 3]
```diff
@@ -1424,6 +1422,24 @@ void MayaHydraSceneIndex::_RemoveRenderItem(const MayaHydraRenderItemAdapterPtr&
...
+void MayaHydraSceneIndex::GetLightedPrimPaths(SdfPathVector& lightedPrimPaths)
+{
+    _MapAdapter<MayaHydraRenderItemAdapter>(
+        [&](MayaHydraRenderItemAdapter* a) {
+            if (a->GetPrimitive() == MHWRender::MGeometry::Primitive::kTriangles) {
+                lightedPrimPaths.emplace_back(a->GetID());
+            }
+        },
+        _renderItemsAdapters);
+    _MapAdapter<MayaHydraDagAdapter>(
```
Comment (lilike-adsk) [inline lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp]: This is for meshAdapter

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.h [hunk 4]
```diff
@@ -305,16 +312,6 @@ class MAYAHYDRALIB_API MayaHydraSceneIndex : public HdRetainedSceneIndex, public
     void _Destroy();
 
 private:
-    // ------------------------------------------------------------------------
```
Comment (lilike-adsk) [inline lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.h:308]: remove dead code

**Comment by lilike-adsk** [comment]
@lanierd-adsk the change is updated, you can start review now, thanks!

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp [hunk 5]
```diff
@@ -1424,6 +1422,24 @@ void MayaHydraSceneIndex::_RemoveRenderItem(const MayaHydraRenderItemAdapterPtr&
...
+        [&](MayaHydraDagAdapter* a) {
+            if (a->HasType(HdPrimTypeTokens->mesh)) {
```
Comment (lanierd-adsk) [inline lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp]: I think this is too restrictive, we should include all Renderable Prims with something like :
// This includes implicit primitive types.
static const VtArray<TfToken> GEOMETRY_PRIM_TYPES = {
    HdRprimTypeTokens->allTokens.begin(),
    HdRprimTypeTokens->allTokens.end()
};

For example, volumes should receive the lighting, I think also curves and points if they support it.
I would put this in a function so we can update it if we want like 
 const VtArray<TfToken> GetLightedPrimitivesType()

**Comment by lanierd-adsk** [comment]
This is only for maya native data right ? What about custom data producers prims and usd prims ?
If we collect the data to be lighted by prim types, could we use a filtering scene index to traverse the whole scene prims (usd+maya+custom) and collect the lighted prims ?

**Review by lanierd-adsk — CHANGES_REQUESTED** [review]

**Comment by lilike-adsk** [comment]
> This is only for maya native data right ? What about custom data producers prims and usd prims ? If we collect the data to be lighted by prim types, could we use a filtering scene index to traverse the whole scene prims (usd+maya+custom) and collect the lighted prims ?

Yes, the original "Lighted" hierarchy is only for native data.  It could be possible to use a SI to support shadows for all data, put the idea in USD shadow ticket: https://jira.autodesk.com/browse/HYDRA-1269

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp (hunk 5 above)
Comment (lilike-adsk) [reply lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp]: Good idea, I'll try it

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp [hunk 6]
```diff
@@ -237,25 +235,6 @@ namespace {
         return sdfDagPath.AppendPath(inPath);
     }
 
-    ///Returns false if this object should not be lighted, true if it should be lighted
```
Comment (lilike-adsk) [inline lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp:240]: The logics are moved to adapter::WantBeLighted() function

File: lib/mayaHydra/hydraExtensions/adapters/adapter.h [hunk 7]
```diff
@@ -117,6 +117,8 @@ class MayaHydraAdapter
     MAYAHYDRALIB_API
     virtual GfVec4f GetDisplayColor() const { return {1.f,1.f,1.f,1.f}; }
 
+    virtual bool WantBeLighted() const { return false; }
```
Comment (lanierd-adsk) [inline lib/mayaHydra/hydraExtensions/adapters/adapter.h]: Nice to have this change, but I would rather call it Illuminated()

**Review by lanierd-adsk — CHANGES_REQUESTED** [review]

**Review by lanierd-adsk — APPROVED** [review]

**Pull request merged**